requirements.txt

```diff
@@ -1,2 +1,3 @@
 torch >= 1.12.0.dev0
-packaging >= 21.3
\ No newline at end of file
+packaging >= 21.3
+numba >= 0.55.2
```

Comment:
Hmm I'm not sure if we want to add a top-line dependency to PiPPy for this experimental feature. How does the performance compare without numba?

Reply:
Base implementation (i.e. their impl) sees a [250x difference](https://github.com/alpa-projects/alpa/blob/9a215e9d49a5b0fbb6e8f2f4f7cc4ab44ff33009/tests/pipeline_parallel/test_dynamic_programming.py#L49). For mine probably the gap is narrower but still.

Reply:
How about we make the dependency optional:

1) Remove it from requirements.txt
2) in `auto_parallelization.py` do something like this:

```
try:
  from numba import njit
  from numba.typed import List as NumbaList
except ImportError:
  def njit(*args, **kwargs):
    pass
    
  NumbaList = list
```

(I haven't run the above code so might be wrong but you should get the idea)

Reply:
this pretty close to what they have ([maybe_numba_jit](https://github.com/alpa-projects/alpa/blob/2c421ee9677d1487253bc056339fdd673faa969d/alpa/util.py#L1135)) so it occurred to me,  but I was thinking we wanted to be perf conscious. anyway i'll add it.